packager/packager.cc

```diff
@@ -298,15 +296,22 @@ class PackageInfoImpl final : public core::packages::PackageInfo {
             // if we don't have any imports, then we can try adding it
             // either before the first export, or if we have no
             // exports, then right before the final `end`
-            uint32_t exportLoc;
+            int64_t exportLoc;
             if (!exports_.empty()) {
                 exportLoc = exports_.front().fqn.loc.beginPos() - "export "sv.size() - 1;
             } else {
                 exportLoc = loc.endPos() - "end"sv.size() - 1;
             }
+
+            std::string_view file_source = loc.file().data(gs).source();
+
+            // Defensively guard against the first export loc or the package's loc being invalid.
```

Comment:
If this is defensive, should we have an `ENFORCE` and/or a log printed in this case?

Reply:
Great point! I'll add an ENFORCE.